src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -431,6 +437,30 @@ private static BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSy
             }
         }
 
+        private bool HasBlockedINumberConversion(Conversion patternConversion, TypeSymbol inputType)
+        {
+            // We want to block constant and relation patterns that have an input type constrained to or inherited from INumberBase<T>, if we don't
+            // know how to represent the constant being matched against in the input type. For example, `1.0 is 1` will work when written inline, but
+            // will fail if the input type is `INumberBase<T>`. We block this now so that we can make make it work as expected in the future without
+            // being a breaking change.
+
+            if (patternConversion.IsIdentity || patternConversion.IsConstantExpression || patternConversion.IsNumeric)
+            {
+                return false;
+            }
+
+            var iNumberBase = Compilation.GetWellKnownType(WellKnownType.System_Numerics_INumberBase_T);
+
+            if (iNumberBase.IsErrorType())
+            {
+                return false;
```

Comment:
>return false; [](http://example.com/codeflow?start=16&length=13)

So the check is going to fail in case of lookup ambiguity and other lookup failures? Should we simply do a full name check instead of relying on WellKnownType, especially that the type is not used by compiler for any other purpose?

Reply:
> So the check is going to fail in case of lookup ambiguity and other lookup failures?

Yes, I would expect so.

> Should we simply do a full name check instead of relying on WellKnownType, especially that the type is not used by compiler for any other purpose?

I suppose we could, but I'm not sure of the advantage of doing so. This is the standard pattern that we do for these cases, so what would we get by changing it?

Reply:
> This is the standard pattern that we do for these cases, 

There are several different patterns we use in different circumstances. See ```IsGenericOrNonGenericExpressionType```, for example.

In my opinion, usage of WellKnownType pattern is not appropriate in this case, because:
- we are not reporting any error in case the lookup fails when the type(s) actually exists.
- it appears that we are really interested in any type named like that, the ability to lookup the type by using WellKnownType algorithm is not one of the requirements. We are not deciding what type to emit. 

> so what would we get by changing it?

I guess we get the check will start working in more scenarios.

Please add tests for the scenarios outlined in the first message of this thread.